### PR TITLE
RDISCROWD-7659 password logic in ProjectCommonForm

### PR DESCRIPTION
**Pull request by n00rsy**
*Issue number of the reported bug or feature request: [RDISCROWD-7659](https://jira.prod.bloomberg.com/browse/RDISCROWD-7659)*

Refactor https://github.com/bloomberg/pybossa/pull/972 after code review.

### CODE REVIEW

File: pybossa/forms/forms.py [hunk 1]
```diff
@@ -84,6 +84,21 @@ def v(form, field):
...
 
 class ProjectCommonForm(Form):
+    def __init__(self, *args, **kwargs):
+        super(ProjectCommonForm, self).__init__(*args, **kwargs)
+        if current_app.config.get('PROJECT_PASSWORD_REQUIRED'):
+            self.password_required = True
+            self.password.validators = [pb_validator.CheckPasswordStrength(
+                                        min_len=PROJECT_PWD_MIN_LEN,
+                                        special=False,
+                                        password_required=True), validators.Required()]
```
Comment (kbecker42) [inline pybossa/forms/forms.py:94]: Could use:
`password_required=self.password_required`

File: pybossa/forms/forms.py [hunk 2]
```diff
@@ -84,6 +84,21 @@ def v(form, field):
...
+        else:
+            self.password_required = False
+            self.password.validators = [pb_validator.CheckPasswordStrength(
+                                        min_len=PROJECT_PWD_MIN_LEN,
+                                        special=False,
+                                        password_required=False), validators.Optional()]
```
Comment (kbecker42) [inline pybossa/forms/forms.py:100]: Could use:
`password_required=self.password_required`

File: pybossa/forms/forms.py [hunk 3]
```diff
@@ -104,26 +119,9 @@ class ProjectCommonForm(Form):
                                             validators=[validators.Required()], choices=[], default='')
     output_data_class = SelectFieldWithProps(lazy_gettext('Output Data Classification'),
                                              validators=[validators.Required()], choices=[], default='')
-    def __init__(self, *args, **kwargs):
-        Form.__init__(self, *args, **kwargs)
 
-class ProjectForm(ProjectCommonForm):
-
-    def __init__(self, *args, **kwargs):
-        super(ProjectForm, self).__init__(*args, **kwargs)
```
[the remark this answers is not in the document]
Comment (n00rsy) [reply pybossa/forms/forms.py:113]: 1. Constructor on lines 107-108:
We removed the constructor in lines 107-108 and declared a new one starting on line 87 for `ProjectCommonForm`.

2. Constructor on lines 112-113
Removing the `__init__` function from `ProjectForm` will make it default to it's parent class's constructor, `ProjectCommonForm` - exactly what we want.

See https://stackoverflow.com/questions/39488875/parent-constructor-called-by-default

File: pybossa/forms/forms.py (hunk 1 above)
Comment (n00rsy) [reply pybossa/forms/forms.py:93]: removed `self.password_required` entirely via 20656dd2bada84b26900d94c7556fb5969cd715f

File: pybossa/forms/forms.py (hunk 2 above)
Comment (n00rsy) [reply pybossa/forms/forms.py:98]: removed `self.password_required` entirely via 20656dd2bada84b26900d94c7556fb5969cd715f

**Review by kbecker42 — APPROVED** [review]

**Pull request merged**